gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/SettingsCategoryPanel.java

```diff
@@ -83,13 +84,29 @@ public void onInitialize() {
 			this.model.getObject().getGradebookInformation().getCategories().add(stubCategoryDefinition());
 		}
 		
+		final WebMarkupContainer settingsCategoriesPanel = new WebMarkupContainer("settingsCategoriesPanel");
+		//Preserve the expand/collapse state of the panel
+		settingsCategoriesPanel.add(new AjaxEventBehavior("shown.bs.collapse") {
+			@Override
+			protected void onEvent(AjaxRequestTarget ajaxRequestTarget) {
+				settingsCategoriesPanel.add(new AttributeModifier("class", "panel-collapse collapse in"));
+			}
+		});
+		settingsCategoriesPanel.add(new AjaxEventBehavior("hidden.bs.collapse") {
+			@Override
+			protected void onEvent(AjaxRequestTarget ajaxRequestTarget) {
+				settingsCategoriesPanel.add(new AttributeModifier("class", "panel-collapse collapse"));
+			}
+		});
+		add(settingsCategoriesPanel);
+
```

Comment:
All panels are now Wicket containers and the CSS (expanded or collapsed) is stored in the session state.